### PR TITLE
Allow checksum verification with standard tools

### PR DESCRIPTION
**Pull request by jprinet**

### CODE REVIEW

**Review by eskatos — APPROVED** [review]

File: src/main/java/org/gradle/crypto/checksum/Checksum.java [hunk 1]
```diff
@@ -170,20 +177,25 @@ public void generateChecksumFiles(InputChanges inputChanges) throws IOException
...
+                        String content;
+                        if(appendFileNameToChecksum.get()){
+                            content = String.format("%s  %s", hashCode, change.getFile().getName());
+                        } else {
+                            content = hashCode.toString();
+                        }
+                        Files.write(content.getBytes(StandardCharsets.UTF_8), sumFile);
                     } catch (IOException e) {
-                        throw new GradleException("Trouble creating checksum", e);
+                        throw new IllegalStateException("Error while creating checksum", e);
```
Comment (jprinet) [inline src/main/java/org/gradle/crypto/checksum/Checksum.java]: I applied your suggestion @etiennestuder / @runningcode.
Please let me know if this is ok for you 👍

Comment (runningcode) [reply src/main/java/org/gradle/crypto/checksum/Checksum.java]: Which line is the one which can throw the `IOException`?
If it is reading or writing the file, we can make the exception more specific.
If the exception can be thrown while computing the checksum, we should change it to `Error creating checksum`.

File: README.md [hunk 2]
```diff
@@ -32,6 +33,14 @@ Currently, only `SHA256` (default), `SHA384`, `SHA512`, and `MD5` are
 supported. Please file an issue or make a pull request if you need support
 for some other hashing algorithm.
 
+If you want to automate the checksum verification with any of `md5sum`, `sha256sum`, `sha384sum` or `sha512sum` tool,
```
Comment (runningcode) [inline README.md]:
```suggestion
If you want to append the filename to the checksum value, set `appendFileNameToChecksum` to true.
This makes it easier to automate the checksum verification with the `md5sum`, `sha256sum`, `sha384sum` or `sha512sum` tools.
```

**Review by runningcode — COMMENTED** [review]
small comments. otherwise looks good.

File: src/main/java/org/gradle/crypto/checksum/Checksum.java (hunk 1 above)
Comment (jprinet) [reply src/main/java/org/gradle/crypto/checksum/Checksum.java]: 2 lines can actually throw the IOException:
- L189 call to hash()
- L196 call to write()

Comment (runningcode) [reply src/main/java/org/gradle/crypto/checksum/Checksum.java]: Ah, in that case let's keep it as `Error creating checksum`.

File: README.md [hunk 3]
```diff
@@ -32,6 +33,15 @@ Currently, only `SHA256` (default), `SHA384`, `SHA512`, and `MD5` are
 supported. Please file an issue or make a pull request if you need support
 for some other hashing algorithm.
 
+If you want to append the filename to the checksum value, set `appendFileNameToChecksum` to true.
+This makes it easier to automate the checksum verification with the `md5sum`, `sha256sum`, `sha384sum` or `sha512sum` tools.
+set the `appendFileNameToChecksum` flag to `true` (`false` by default):
```
Comment (runningcode) [inline README.md]: This line is extraneous. The default information could be kept though.

Comment (jprinet) [reply README.md]: Good catch

**Review by runningcode — APPROVED** [review]
Looks good, thanks Jerome.

**Pull request merged**